libkbfs/identify_util.go

```diff
@@ -6,12 +6,111 @@ package libkbfs
 
 import (
 	"fmt"
+	"sync"
 
 	"github.com/keybase/client/go/libkb"
 	"github.com/keybase/client/go/protocol/keybase1"
 	"golang.org/x/net/context"
+	"golang.org/x/sync/errgroup"
 )
 
+type extendedIdentify struct {
+	behavior   keybase1.TLFIdentifyBehavior
+	userBreaks chan keybase1.TLFUserBreak
+
+	tlfBreaksLock sync.Mutex
+	tlfBreaks     *keybase1.TLFBreak
+}
+
+func (ei *extendedIdentify) userBreak(username libkb.NormalizedUsername, uid keybase1.UID, breaks *keybase1.IdentifyTrackBreaks) {
+	if ei.userBreaks == nil {
+		return
+	}
+
+	ei.userBreaks <- keybase1.TLFUserBreak{
+		Breaks: breaks,
+		User: keybase1.User{
+			Uid:      uid,
+			Username: string(username),
+		},
+	}
+}
+
+func (ei *extendedIdentify) makeTlfBreaksIfNeeded(numUserInTlf int) {
+	if ei.userBreaks == nil {
+		return
+	}
+
+	b := &keybase1.TLFBreak{}
+	for i := 0; i < numUserInTlf; i++ {
+		ub := <-ei.userBreaks
```

Comment:
This function should take a context and select on `ctx.Done()` here, otherwise it could block forever if the corresponding identify call errors out early.